ui_text_input.js

```diff
@@ -0,0 +1,34 @@
+module.exports = function(RED) {
+    var ui = require('../ui')(RED);
+
+    function TextInputNode(config) {
+        RED.nodes.createNode(this, config);
+        var node = this;
+
+        var group = RED.nodes.getNode(config.group);
+        if (!group) { return; }
+        var tab = RED.nodes.getNode(group.config.tab);
+        if (!tab) { return; }
+
+        var done = ui.add({
+            node: node,
+            tab: tab,
+            group: group,
+            control: {
+                type: config.delay < 0 ? 'text-input-CR' : 'text-input',
```

Comment:
but testing against <0 means     type is   'text-input-CR'   if <0    otherwise  it's  'text-input' -  so 0 means 'text-input'    - the triple is equivalent to   if (config.delay <0) then (text-input-cr) else (text input)   isn't it ?